### PR TITLE
60 implement a new passport auth

### LINKED ISSUE #60: Implement a new Passport auth
**Short description:**
Implement a new way of auth with passport.

**Describe the solution you'd like**
Choose one between Google, Apple and FacebooK.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to add Google and Facebook auth methods.

Fixes #60 

## Added features

* The Google auth methods is now available !
 * Go to /api/login/google and register yourself.
 * Generate a valid bearer token
 * Auto-validate E-mail
 * Fill all your account information with Google one's

* The Facebook auth methods is now available !
 * Go to /api/login/facebook
 * Generate a valid bearer token
 * Auto-validate E-mail
 * Fill all your account information with Facebook one's

* Link made with Facebook and Google auth (Creation of an account with Google and login with Facebook is possible ! waw)

* Update user database information for including Google Id and Facebook Id (Auth use)

### CODE REVIEW

File: server/database.env [hunk 1]
```diff
@@ -4,4 +4,8 @@ password=
 database=
 DATABASE_URL=
 JWT_SECRET=eyJhbGciOiJIUzI1NiJ9.eyJSb2xlIjoiQWRtaW4iLCJJc3N1ZXIiOiJJc3N1ZXIiLCJVc2VybmFtZSI6IkphdmFJblVzZSIsImV4cCI6MTY3NDEyNjIwOSwiaWF0IjoxNjc0MTI2MjA5fQ.2vxaSmmfQ14Ol4Bgy3Cp62N5HbHBAnqlAkBDd
-JWT_EXPIRE=10
\ No newline at end of file
+JWT_EXPIRE=10
+GMAIL_CLIENT_ID=770124443966-6do9f24ebp54m.apps.googleusercontent.com
```
Comment (GuyomT) [inline server/database.env:8]: SECRETS ??!

File: server/passport/facebook.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const database = require('../database_init')
+const { hash, compare } = require('../utils')
+
+/**
+ * Strategy options needed by passport
+ */
+const options = {
+  clientID: process.env.FACEBOOK_CLIENT_ID,
+  clientSecret: process.env.FACEBOOK_CLIENT_SECRET,
+  callbackURL: 'http://localhost:8080/api/login/facebookCallBack',
```
Comment (GuyomT) [inline server/passport/facebook.js:14]: Maybe add a comment to not forget it is a MVP hyperlink (localhost:8080 may change)

File: server/passport/google.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const database = require('../database_init')
+const { hash, compare } = require('../utils')
+
+/**
+ * Strategy options needed by passport
+ */
+const options = {
+  clientID: process.env.GMAIL_CLIENT_ID,
+  clientSecret: process.env.GMAIL_CLIENT_SECRET,
+  callbackURL: 'http://localhost:8080/api/login/googleCallBack',
```
Comment (GuyomT) [inline server/passport/google.js:14]: Same as before

**Review by GuyomT — COMMENTED** [review]
We can't push secrets

**Review by GuyomT — DISMISSED** [review]
LGTM, got pranked

File: server/database.env (hunk 1 above)
Comment (LucasTesnier) [reply server/database.env:8]: It's not the real one, don't worry. It's for Unit Testing ;)

File: server/passport/facebook.js (hunk 2 above)
Comment (LucasTesnier) [reply server/passport/facebook.js:14]: Good idea

File: server/passport/google.js (hunk 3 above)
Comment (LucasTesnier) [reply server/passport/google.js:14]: Good idea

**Review by GuyomT — DISMISSED** [review]
LGTM !!

**Review by TomDUVAL-MAHE — DISMISSED** [review]
ntr

File: server/server_app.js [hunk 4]
```diff
@@ -4,16 +4,31 @@ const express = require('express')
...
+const auth_facebook = require('./passport/facebook')
 const utils = require('./utils')
 const gmail = require('./services/gmail/reactions/send_email')
 const jwt = require('jwt-simple')
 const { hash } = require('./utils')
 require('dotenv').config({ path: '../database.env' })
 
 const app = express()
+
+passport.serializeUser(function (user, done) {
```
Comment (tbellicha) [inline server/server_app.js]: use `=>` instead of `function`

File: server/server_app.js [hunk 5]
```diff
@@ -4,16 +4,31 @@ const express = require('express')
...
+  done(null, user.id)
+})
+
+passport.deserializeUser(function (id, done) {
```
Comment (tbellicha) [inline server/server_app.js]: use `=>` instead of `function`

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by ZQUEMA — DISMISSED** [review]

File: server/server_app.js (hunk 4 above)
Comment (LucasTesnier) [reply server/server_app.js]: Done

Comment (LucasTesnier) [reply server/server_app.js]: Done

File: server/server_app.js (hunk 5 above)
Comment (LucasTesnier) [reply server/server_app.js]: Done

**Review by ZQUEMA — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**